File: frontend/src/types/lead.ts
export interface LeadEmail {
  email: string;
  is_valid: boolean | null;
  is_primary: boolean;
}

export interface LeadPhone {
  phone_raw: string | null;
  phone_normalized: string | null;
  phone_type: string;
  is_valid: boolean | null;
  is_primary: boolean;
}

export interface Lead {
  id: string;
  lead_kind: "b2b" | "b2c";
  company_name: string;
  siren: string | null;
  naf_code: string | null;
  sector: string | null;
  website: string | null;
  address: string | null;
  postal_code: string | null;
  city: string | null;
  department: string | null;
  region: string | null;
  quality_score: number;
  source: string;
  source_url: string | null;
  is_duplicate: boolean;
  consent_status: ConsentStatus;
  emails: LeadEmail[];
  phones: LeadPhone[];
  extraction_job_id: string;
  organization_id: string;
  created_at: string;
}

export interface LeadFilters {
  lead_kind?: "b2b" | "b2c";
  search?: string;
  sector?: string;
  city?: string;
  department?: string;
  region?: string;
  min_score?: number;
  max_score?: number;
  date_from?: string;
  date_to?: string;
  has_email?: boolean;
  has_phone?: boolean;
  is_duplicate?: boolean;
  consent_granted_only?: boolean;
  source?: string;
  page?: number;
  page_size?: number;
  ordering?: string;
}

export interface SuggestedSegment {
  code: string;
  label: string;
  description: string;
  count: number;
  filters: Record<string, string | number | boolean>;
}

export type ConsentStatus = "granted" | "denied" | "revoked" | "unknown";
export type ConsentScope = "email" | "phone" | "sms" | "whatsapp" | "all";
export type LawfulBasis = "consent" | "contract" | "legitimate_interest";

export interface LeadConsent {
  id: string;
  lead_id: string;
  organization_id: string;
  consent_status: ConsentStatus;
  consent_scope: ConsentScope;
  consent_source: string | null;
  consent_at: string | null;
  consent_text_version: string | null;
  consent_proof_ref: string | null;
  privacy_policy_version: string | null;
  lawful_basis: LawfulBasis;
  source_campaign: string | null;
  source_channel: string | null;
  ip_hash: string | null;
  user_agent_hash: string | null;
  double_opt_in: boolean;
  double_opt_in_at: string | null;
  purpose: string | null;
  data_retention_until: string | null;
  created_at: string;
  updated_at: string;
}

export interface LeadConsentUpdatePayload {
  consent_status?: ConsentStatus;
  consent_scope?: ConsentScope;
  consent_source?: string | null;
  consent_at?: string | null;
  consent_text_version?: string | null;
  consent_proof_ref?: string | null;
  privacy_policy_version?: string | null;
  lawful_basis?: LawfulBasis;
  source_campaign?: string | null;
  source_channel?: string | null;
  ip_hash?: string | null;
  user_agent_hash?: string | null;
  double_opt_in?: boolean;
  double_opt_in_at?: string | null;
  purpose?: string | null;
  data_retention_until?: string | null;
}

export interface B2CLeadIntakePayload {
  full_name: string;
  email?: string | null;
  phone?: string | null;
  city?: string | null;
  consent_source: "web_form" | "meta_lead_ads" | "google_lead_form" | "partner_api" | "crm_import";
  consent_at: string;
  consent_text_version: string;
  consent_proof_ref: string;
  privacy_policy_version: string;
  source_campaign?: string | null;
  source_channel?: "web" | "facebook" | "instagram" | "google" | "tiktok" | "partner" | "import" | null;
  purpose?: string | null;
  double_opt_in?: boolean;
  double_opt_in_at?: string | null;
}

File: backend/app/schemas/lead.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class LeadEmailSchema(BaseModel):
    email: str
    is_valid: bool | None = None
    is_primary: bool = False

    model_config = {"from_attributes": True}


class LeadPhoneSchema(BaseModel):
    phone_raw: str | None = None
    phone_normalized: str | None = None
    phone_type: str = "unknown"
    is_valid: bool | None = None
    is_primary: bool = False

    model_config = {"from_attributes": True}


class LeadResponse(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    extraction_job_id: uuid.UUID | None = None
    company_name: str
    siren: str | None = None
    naf_code: str | None = None
    sector: str | None = None
    website: str | None = None
    address: str | None = None
    postal_code: str | None = None
    city: str | None = None
    department: str | None = None
    region: str | None = None
    quality_score: int = 0
    source: str
    source_url: str | None = None
    lead_kind: Literal["b2b", "b2c"] = "b2b"
    is_duplicate: bool = False
    is_similar: bool = False
    consent_status: Literal["granted", "denied", "revoked", "unknown"] = "unknown"
    emails: list[LeadEmailSchema] = []
    phones: list[LeadPhoneSchema] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LeadUpdate(BaseModel):
    company_name: str | None = None
    sector: str | None = None
    website: str | None = None
    address: str | None = None
    postal_code: str | None = None
    city: str | None = None


class LeadFilters(BaseModel):
    min_score: int | None = None
    max_score: int | None = None
    extraction_job_id: uuid.UUID | None = None
    sector: str | None = None
    city: str | None = None
    department: str | None = None
    region: str | None = None
    source: str | None = None
    lead_kind: Literal["b2b", "b2c"] | None = None
    has_email: bool | None = None
    has_phone: bool | None = None
    is_duplicate: bool | None = None
    is_similar: bool | None = None
    consent_granted_only: bool | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None
    search: str | None = None


class SuggestedSegment(BaseModel):
    code: str
    label: str
    description: str
    count: int
    filters: dict[str, str | int | bool]


class SuggestedSegmentResponse(BaseModel):
    items: list[SuggestedSegment]
